api/src/main/proto/stellarstation/api/v1/radio/radio.proto

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 Infostellar, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package stellarstation.api.v1.radio;
+
+option go_package = "radio";
+
+option java_multiple_files = true;
+option java_outer_classname = "RadioProto";
+option java_package = "com.stellarstation.api.v1.radio";
+
+// Messages for modeling radio device (e.g., receivers, transmitters) configuration in API messages.
+
+// A radio device for RF communication. Usually, a satellite will have one or more radio devices for
+// communicating with the ground and a ground station will have one or more radio devices for
+// communicating with the satellite. Such devices may not actually correspond to actual hardware
+// devices, i.e., in the case of Software Defined Radio (SDR).
+message RadioDeviceConfiguration {
+  // The center frequency of the device, in Hz.
+  uint64 center_frequency_hz = 1;
+
+  // The protocol used by this device when doing RF communication.
+  CommunicationProtocol protocol = 2;
+}
+
+// A communication protocol used with a radio device. These must contain all the parameters
+// needed to configure the device for use either in transmission or reception.
+message CommunicationProtocol {
+  // The type of modulation used by this radio device.
+  Modulation modulation = 1;
+
+  // The type of framing used for communication on the device. If `bitstream`, the device is only
+  // demodulating / modulating without applying any higher-level framing (i.e., packeting) protocol.
+  oneof Framing {
+    BitStream bitstream = 2;
+
+    AX25 ax25 = 3;
+  }
+}
+
+// Communication of a stream of bits with no framing defined.
+message BitStream {
+}
+
+// PACKET communication based on AX.25. https://www.sigidwiki.com/wiki/PACKET.
+message AX25 {
+  // Bit rate, e.g. 9600 for FSK9600, 1200 for AFSK1200.
+  uint64 bit_rate = 1;
```

Comment:
Two things regarding this field in the AX25 data type:

1, major) bitrate is a parameter of the modulator that should be part of the radio configuration object. AX.25 is just an encoding format that runs on top of several modulations with different parameters, including different bitrates. Please use the following [https://www.tapr.org/pdf/AX25.2.2.pdf] as a reference.

2, minor) it is more common to see "bitrate" alltogether in terms of communications naming conventions.

Reply:
Thanks - can you take another look to see if the fieldname / comment make sense or could be more clear? @rtubio @reiinakano

Reply:
The rest of the message LGTM.

Reply:
The comment makes sense, although I would refer to the official specification for AX.25 better. We just need to remove the reference to "bit_rate" within the AX.25 protocol header.

Reply:
I don't understand the comment - I think you might be looking at the old code instead of new (AX25 header doesn't have bitrate anymore). If that's not the case, let me know since I'm missing something.